skbio/sequence/_sequence.py

```diff
@@ -887,18 +808,7 @@ def __eq__(self, other):
         if self.__class__ != other.__class__:
             return False
 
-        # we're not simply comparing self.metadata to other.metadata in order
-        # to avoid creating "empty" metadata representations on the sequence
-        # objects if they don't have metadata. same strategy is used below for
-        # positional metadata
-        if self.has_metadata() and other.has_metadata():
-            if self.metadata != other.metadata:
-                return False
-        elif not (self.has_metadata() or other.has_metadata()):
-            # both don't have metadata
-            pass
-        else:
-            # one has metadata while the other does not
+        if not super(Sequence, self).__eq__(other):
```

Comment:
Nice! I was wondering what a good way to do this without explicitly depending on the implementation of the mixin would be.